scripts/variantstore/wdl/extract/hail_create_vat_inputs.py

```diff
@@ -73,19 +73,70 @@ def remove_too_many_alt_allele_sites(vds):
 
 
 
+def annotate_adj(
+        mt: hl.MatrixTable,
+        adj_gq: int = 30,
+        adj_ab: float = 0.2,
+) -> hl.MatrixTable:
+    """
+    Annotate genotypes with adj criteria (assumes diploid).
+    Defaults correspond to gnomAD values.
+    """
+    if "GT" not in mt.entry and "LGT" in mt.entry:
+        print("No GT field found, using LGT instead.")
+        gt_expr = mt.LGT
+    else:
+        gt_expr = mt.GT
+    if "AD" not in mt.entry and "LAD" in mt.entry:
+        print("No AD field found, using LAD instead.")
+        ad_expr = mt.LAD
+    else:
+        ad_expr = mt.AD
+    return mt.annotate_entries(
+        adj=get_adj_expr(
+            gt_expr, mt.GQ, ad_expr, adj_gq, adj_ab
+        )
+    )
+def get_adj_expr(
+    gt_expr: hl.expr.CallExpression,
+    gq_expr: Union[hl.expr.Int32Expression, hl.expr.Int64Expression],
+    ad_expr: hl.expr.ArrayNumericExpression,
+    adj_gq: int = 30,
+    adj_ab: float = 0.2,
+) -> hl.expr.BooleanExpression:
+    """
+    Get adj genotype annotation.
+    Defaults correspond to gnomAD values.
+    """
+    return (
+        (gq_expr >= adj_gq)
+        & (
+            hl.case()
+            .when(~gt_expr.is_het(), True)
+            .when(gt_expr.is_het_ref(), ad_expr[gt_expr[1]] / hl.sum(ad_expr) >= adj_ab)
```

Comment:
If AD is only for non-refs, wont this  `ad_expr[gt_expr[1]] / hl.sum(ad_expr)` always eval to 1? Seems like the AB bit for het ref is slightly misleading. Sorry if this has other use cases though -- just going off the slack thread.

Reply:
Slack thread was somewhat shorthand. AB is not available for hom-ref sites, but it has ref allele computed correctly for hets, I believe

Reply:
Slack thread was somewhat shorthand. AB is not available for hom-ref sites, but it has ref allele computed correctly for hets, I believe

Reply:
Ah, just reread the thread and read it too fast the first time. Would it make sense to add DP to adj with a conditional on hom-ref returning True but then running `hl.sum(ad_expr)` for non-refs? I guess adj would mean slightly different things within the dataset, but you could be a bit more confident in your rare adj ACs, nothing really changes for ANs....Though I guess youd want different GQ cutoffs for hom-ref and non-ref then. Sorry this may be more slack thread material.